### ISSUE TITLE
px.timeline chart incorrectly displays timeframes of 10 seconds and smaller.

### ISSUE THREAD

**Issue by Denpeer** [plotly/plotly.py#2658]
The following piece of python code that uses a ISO-formatted date string produces the graph below.
```
import plotly.express as px
import pandas as pd

bars = pd.DataFrame([{'Task': 'kitter_pick', 'Start': '2020-07-22T23:09:18.221013', 'Finish': '2020-07-22T23:09:25.221013', 'Counter': 2465}])

fig = px.timeline(bars, x_start="Start", x_end="Finish", y="Task",hover_name='Counter')
fig.update_yaxes(autorange="reversed")  # otherwise tasks are listed from the bottom up
fig.show()
```
[result][1]
The snippet is similar to https://plotly.com/python/time-series/ from the plotly documentation.

The timescale is completely incorrect due to the finish attribute being incorrectly interpreted by plotly.

When changing the timestrings slightly you obtain the correct result:
```
bars = pd.DataFrame([{'Task': 'kitter_pick', 'Start': '2020-07-22T23:09:18.221013', 'Finish': '2020-07-22T23:10:25.221013', 'Counter': 2465}])
```
[correct result][2]

Does anybody know why this happens and if there is a way around this?

I have also tried:
* datetime.datetime objects
* np.datetime64 objects

Both gave the same results

[stackoverflow link](https://stackoverflow.com/questions/63035572/why-does-plotly-incorrectly-interpret-this-iso-datetime-string)

  [1]: https://i.stack.imgur.com/SDmeV.png
  [2]: https://i.stack.imgur.com/c5q6z.png

**Comment by Talon24** [comment]
There is also a segment when the (end date is smaller than start + 1 second) and (end date is larger or equal than start - 0.001 seconds) where the chart does work as expected.

**Comment by arielcedola** [comment]
Got the same issue with timeline here. The problem is when the timedelta is less than 10 seconds, although when x_start = x_end it works fine.

**Comment by nicolaskruchten** [comment]
Thanks for reporting this! It's indeed a bug in the underlying Javascript library, which we will fix: https://github.com/plotly/plotly.js/issues/5057

**Comment by nicolaskruchten** [comment]
Should be fixed in the next version of `plotly`, within 10 days.

**Comment by nicolaskruchten** [comment]
Fixed on master.

**Comment by nicolaskruchten** [comment]
Resolved in 4.10 https://community.plotly.com/t/announcing-plotly-py-4-10-date-axis-improvements-a-faster-px-imshow-and-full-figures-for-development/44685

**Comment by dpasqualin** [comment]
@nicolaskruchten I'm not sure if this issue is entirely solved, could you please see whether #2518 is related? Thank you.

**Comment by saksham7778** [comment]
@nicolaskruchten Can we use start and end to view information in nanoseconds ( or by passing an integer array ) ? Please help
